### PR TITLE
fix: response-rewrite plugin might cause Apache AIPSIX hanging

### PR DESCRIPTION
**Pull request by Hazel6869**
### Description

Fixes  https://github.com/apache/apisix/issues/7722

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by spacewander** [comment]
Could you use `Fixes #7722` in the comment message?

File: t/plugin/response-rewrite.t [hunk 1]
```diff
@@ -699,3 +699,50 @@ X-A: 127.0.0.1
 X-B: from 127.0.0.1 to 127.0.0.1:1980
 --- no_error_log
 [error]
+
+
+
+=== TEST 25: set empty body
```
Comment (spacewander) [inline t/plugin/response-rewrite.t:705]: I run your test locally, and it can be passed even without newly added modification.

Comment (Hazel6869) [reply t/plugin/response-rewrite.t:705]: Are there any relevant examples? We tested and found that the original code would go wrong when running this case

Comment (Hazel6869) [reply t/plugin/response-rewrite.t:705]: @spacewander  hi you can see https://github.com/Hazel6869/apisix/runs/8244062115?check_suite_focus=true

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**